main.py

```diff
@@ -195,16 +199,71 @@ def has_role(user, role_name):
                     except Exception as exc:
                         await log_msg(f'Could not send role set message to {usr.mention}! {exc}')
             logger.info(f'Done, sleeping...')
-            await asyncio.sleep(20)  # task runs every 10 seconds
+            await asyncio.sleep(20)  # task runs every 20 seconds
         except Exception as e:
-            log_msg(f'Error checking spreadsheet: {e}')
-
+            await log_msg(f'Error checking spreadsheet: {e}')
 
 async def send_welcome(user: discord.Member):
     welcome_message = sheet2.col_values(1)[1]
     await client.send_message(user, welcome_message)
     logger.info(f'Sent welcome message to {user}')
 
+#Ask if newly creted role is game role.
+@client.event
+async def on_server_role_create(new_role):
+    if new_role.name != 'new role':
```

Comment:
You should make one function (might not have to be async) that takes in a new role:
```python
def on_role_modify(new_role):  # or whatever you want to call it
    if new_role.name != 'new role':
        # blah....
```
and use that function in both `on_server_role_create` and `on_server_role_update`, with the same logic (just look at where the code repeats and replace that entire block).